ui/button.mod/button.js

```diff
@@ -152,22 +152,35 @@ const Button = (exports.Button = class Button extends Control {
         return this._promise;
     }
 
-    set promise(value) {
-        const handleResolution = () => {
-            if (handleResolution.promise === this._promise) {
-                this.classList.remove("mod--pending");
-                this._promise = undefined;
-            }
-        };
+    set promise(promise) {
+        // Only proceed if the new promise is different from the current one
+        if (this._promise === promise) return;
+
+        const shouldClearPendingState = !!this._promise;
+        this._promise = promise;
+
+        if (promise) {
+            // Set up pending state when promise is set
+            this.classList.add("mod--pending");
+
+            // Create a closure to handle promise resolution
+            const handleResolution = () => {
```

Comment:
would be easier to use a regular function for handleResolution()  here that would memorize a local originalPromise in set promise() than using a fat arrow one and storing 
            handleResolution.originalPromise = promise;
on it, no?